frame/staking/src/lib.rs

```diff
@@ -1164,23 +1017,73 @@ decl_storage! {
 pub mod migrations {
 	use super::*;
 
-	#[derive(Decode)]
-	struct OldValidatorPrefs {
-		#[codec(compact)]
-		pub commission: Perbill
-	}
-	impl OldValidatorPrefs {
-		fn upgraded(self) -> ValidatorPrefs {
-			ValidatorPrefs {
-				commission: self.commission,
-				.. Default::default()
+	pub mod v6 {
+		use super::*;
+		use frame_support::{traits::Get, weights::Weight, pallet_prelude::*};
+
+		pub trait V6Config: crate::Config {
+			/// The storage prefix of the pallet set by the outer runtime declaration.
+			type PalletPrefix: Get<&'static str>;
+		}
+
+		// TODO: complete this
+		macro_rules! generate_storage_types {
```

Comment:
What are the implications of leaving this macro as-is?

Reply:
well, there's an unreachable arm that can panic for starters :D my pan is to move it maybe somewhere that it can be reused? cc @thiolliere